### PR TITLE
Add new NUC model and prune outdated advice.

### PR DESCRIPTION
**Pull request by justintroutman**
## Status

Ready for review!

## Description of Changes

Changes NUC recommendation to NUC5i5MYHE and indicates that the previously recommended D54250WYK is now at End of Life / End of Interactive Support; also, removed some outdated, inapplicable information regarding antiquated SD versions.

### If you made changes to documentation:

- [x] Doc linting passed locally

### CODE REVIEW

File: docs/hardware.rst [hunk 1]
```diff
@@ -263,62 +263,41 @@ Specific Hardware Recommendations
...
+    and `End of Interactive Support` statuses. If you're currently using this
+    model for your SecureDrop setup, and need hardware support, you'll need to
+    consult the `support community <https://communities.intel.com/community/tech/nuc>`__ forum.
+
+.. note:: If you encounter issues booting Ubuntu on the NUCs, try
 	  updating the BIOS according to `these instructions
 	  <http://arstechnica.com/gadgets/2014/02/new-intel-nuc-bios-update-fixes-steamos-other-linux-booting-problems/>`__.
 
-.. note:: Some BIOS versions on the NUC will cause the server to
+.. caution:: Some BIOS versions on the NUC will cause the server to
```
Comment (dachary) [inline docs/hardware.rst]: The thread below is marked **Answered** and one of the [latest comments](https://communities.intel.com/message/432692#432692) dated november 2016 point to [the release notes of the BIOS version fixing the issue](https://downloadmirror.intel.com/26263/eng/RY_0359_ReleaseNotes.pdf). Although the associated comment is terse (**Fixed an issue where the NUC can't wake up from S3 by pressing the power button in Linux.**), it looks this warning can be removed.

Comment (justintroutman) [reply docs/hardware.rst]: Excellent! Thanks for the update on that issue.

**Comment by justintroutman** [comment]
@dachary Per your review, I updated the note to indicate that the problem regarding suspension, when using older NUC BIOS versions, has since been resolve; links to the Intel community forum and the relevant release notes were added for those who might still be running old hardware.

**Comment by redshiftzero** [comment]
Since maintainer @dachary and documentation reviewer @edenemmanuel have approved this, I'm merging.

**Pull request merged**